.github/workflows/byob-gradle-trw.yml

```diff
@@ -0,0 +1,78 @@
+# Copyright 2023 SLSA Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+
+on:
+  workflow_call:
+    inputs:
+      name1:
+        description: "name 1"
+        required: false
+        type: string
+      name2:
+        description: "name 2"
+        required: false
+        type: string
+      rekor-log-public:
+        description: "private"
+        required: false
+        type: boolean
+        default: false
+      provenance-overwrite:
+        description: "overwrite provenance if already present"
+        required: false
+        type: boolean
+        default: false
+      release-tag: #test
+        description: "tag collision with dispatch event"
+        required: false
+        type: string
+        default: "v100"
+      artifact-list:
+        description: 'comma-separated list of paths to artifacts to be released. Use GRADLE_VERSION for the gradle version'
+        type: string
+        required: true
+      jdk-version:
+        description: 'The JDK version'
+        required: false
+        default: 17
+        type: number
+jobs:
+  slsa-setup:
+    permissions:
+      id-token: write # For token creation.
+    outputs:
+      slsa-token: ${{ steps.generate.outputs.slsa-token }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Generate the token
+        id: generate
+        uses: slsa-framework/slsa-github-generator/actions/delegator/setup-token@main
+        with:
+          slsa-workflow-recipient: "delegator_generic_slsa3.yml"
+          slsa-rekor-log-public: ${{ inputs.rekor-log-public }}
+          slsa-runner-label: "ubuntu-latest"
+          slsa-build-action-path: "./internal/builders/gradle"
+          slsa-workflow-inputs: ${{ toJson(inputs) }}
+  slsa-run:
+    needs: [slsa-setup]
+    permissions:
+      id-token: write # For signing.
```

Comment:
Note: we made some changes to the nodejs builder and it now only requires `id-token` to be `write`. This is based on feedback. See https://github.com/slsa-framework/slsa-github-generator/blob/main/.github/workflows/builder_nodejs_slsa3.yml#L101-L109

So question: do you think we should follow the same design? The PW is in charge of uploading to the registry, which means they can use the builder and *not* publish (e.g., they only want to add it to GH releases), and also they don't need to give us the permissions (this was the feedback we received). They may also download the provenance and upload themselves from a local machine without the need to share their password with GitHub.

For nodejs, we provide an Action to publish the provenance to the registry for simplicity. This is a temporary solution until gradle CLI would be able to upload the provenance via a CLI (for npm, this is not supported it yet either; but once it is, users won't need our Actions. See https://github.com/slsa-framework/slsa-github-generator/tree/main/internal/builders/nodejs#getting-started that explains the use of our temporary Action for publication)

Reply:
not of a PW, but the nodejs builder does something similar and the steps should be the same. https://github.com/slsa-framework/slsa-github-generator/blob/main/actions/nodejs/publish/action.yml is the Action that download the attestation(s) and uploads it.
You can use the nodejs secure-attestations-download Action to tests, and then we can switch to using https://github.com/slsa-framework/slsa-github-generator/pull/2136 when it's merged.

Reply:
Changed the permission of "contents" and the delegator to the lowperm one. I will need to add the publisher but would prefer to add that once the builder is merged.

Reply:
let me know when the PR is ready for final review

Reply:
This PR can get reviewed with the purpose of getting it merged.

Reply:
I have added a tracker here: https://github.com/slsa-framework/slsa-github-generator/issues/2154